src/app/pages/client/explore/Explore.tsx

```diff
@@ -37,40 +41,41 @@ import { AsyncStatus, useAsyncCallback } from '../../../hooks/useAsyncCallback';
 import { useNavToActivePathMapper } from '../../../hooks/useNavToActivePathMapper';
 import { PageNav, PageNavContent, PageNavHeader } from '../../../components/page';
 import { stopPropagation } from '../../../utils/keyboard';
+import { useExploreServers } from '../../../hooks/useExploreServers';
 
-export function AddServer() {
+type AddExploreServerPromptProps = {
+  onSubmit: (server: string) => Promise<void>;
+  header: ReactNode;
+  children: ReactNode;
+  selected?: boolean;
+};
+export function AddExploreServerPrompt({
+  onSubmit,
+  header,
+  children,
+  selected = false,
+}: AddExploreServerPromptProps) {
   const mx = useMatrixClient();
-  const navigate = useNavigate();
   const [dialog, setDialog] = useState(false);
   const serverInputRef = useRef<HTMLInputElement>(null);
 
-  const [exploreState] = useAsyncCallback(
-    useCallback((server: string) => mx.publicRooms({ server, limit: 1 }), [mx])
-  );
-
   const getInputServer = (): string | undefined => {
     const serverInput = serverInputRef.current;
     if (!serverInput) return undefined;
     const server = serverInput.value.trim();
     return server || undefined;
   };
 
-  const handleSubmit: FormEventHandler<HTMLFormElement> = (evt) => {
-    evt.preventDefault();
-    const server = getInputServer();
-    if (!server) return;
-    // explore(server);
+  const [submitState, handleSubmit] = useAsyncCallback(
+    useCallback(async () => {
+      const server = getInputServer();
+      if (!server) return;
 
-    navigate(getExploreServerPath(server));
-    setDialog(false);
-  };
-
-  const handleView = () => {
-    const server = getInputServer();
-    if (!server) return;
-    navigate(getExploreServerPath(server));
-    setDialog(false);
-  };
+      await mx.publicRooms({ server, limit: 1 });
```

Comment:
i think we should not check for public rooms here(as it delay the input) and adds whatever input as server, user will see the error when they view the server.